### DISCUSSION TITLE
Sparkle properties in Preference window now not showing.

### DISCUSSION THREAD

**Discussion by jchaplin** [sparkle-project/Sparkle#2403] [Q&A]
Hi all,

I have bound properties in this preference window that link a text label to "last update check date" and a button enabled state on whether the app "can check for updates". ie:
- updater.lastUpdateCheckDate
- updater.canCheckForUpdates

If I disable the binding, the button does the check fine (so the SPUStandardUpdaterController object) is working
but the dates and bindings no longer work. 

![CleanShot 2023-07-04 at 12 33 56@2x](https://github.com/sparkle-project/Sparkle/assets/85207/56cdf2f7-f2bc-4dde-b45d-8ff956626b4a)

As you can see, the same bindings work for the App menu
<img width="264" alt="CleanShot 2023-07-04 at 12 56 37@2x" src="https://github.com/sparkle-project/Sparkle/assets/85207/3babf43c-6a94-40bc-bb03-839839225f6b">

**Comment by zorgiepoo** [comment]
Maybe make sure the `updater` in question isn't `nil` and is hooked up correctly? If you have the `SPUStandardUpdaterController` in there as your "top level instance" you may need to do something like `updaterController.updater.canCheckForUpdates`. Is this SwiftUI or Cocoa? https://sparkle-project.org/documentation/preferences-ui/ is our guide for this.

**Reply by jchaplin** [reply]
it was working in older releases and seems to be hooked up ok.
If I disable the "enabled" binding the button works. 

<img width="1600" alt="CleanShot 2023-07-04 at 14 43 40@2x" src="https://github.com/sparkle-project/Sparkle/assets/85207/5938b6af-7335-4966-8ac0-be0c468c948b">

**Reply by zorgiepoo** [reply]
I think I can reproduce this. I'll have to investigate further.. It's strange, KVO on the `SPUUpdater.canCheckForUpdates` property works but something perhaps with the additional level of indirection with the updater controller makes it not work here. And somehow this broke. (the NSMenuItem path is different validation-wise by the way, that doesn't use KVO)

**Reply by jchaplin** [reply]
It is the KVO Bindings that done seem to work. But the Sent Action works find with the Updater Controller

**Reply by zorgiepoo** [reply]
It works programmatically without the additional level of indirection

```objc
self.observer = self.updaterController.updater.observe(\SPUUpdater.canCheckForUpdates, options: .new) { updater, change in
            print("Can check now: \(change.newValue)")
        }
```

In Sparkle's Test App, if I add a button and bind its Enabled to `self.updater.canCheckForUpdates` in IB where `self` == File Owner, that also works.

**Reply by jchaplin** [reply]
thanks,  but i am unsure where the additional level of indirection is. The Updater Controller object is in the scene.

**Reply by zorgiepoo** [reply]
I was just debugging the issue. The extra level is using KVC/KVO to access the `updater` from the `SPUStandardUpdaterController` and instantiating the updater controller in that same nib in this specific case. I found the issue and I'm fixing it in https://github.com/sparkle-project/Sparkle/pull/2404

Our website documentation examples don't access the updater from the updater controller using KVO when the updater controller may not be fully instantiated, so those examples did not break. 

(On a side note the `SPUStandardUpdaterController` is typically instantiated as part of the main application (with access to MainMenu) -- so it may be a bit strange here to find it instantiated as part of a license scene of sorts).

**Reply by jchaplin** [reply]
Thanks so much and it makes sense. 
I have the SPUStandardUpdaterController also initiated in the main application to get the "check for updates" menu which also has the enabled binding (which works). I might try and set the binding programmatically in the License scene via the App Delegate.

**Reply by zorgiepoo** [reply]
You probably do not want to instantiate the updater controller twice. You shouldn’t have multiple instances running simultaneously. You can probably pass the updater controller to the license scene and it can hook up bindings accordingly referencing the file owner.

**Reply by jchaplin** [reply]
Makes sense thanks for your help.
